src/main/java/com/startstepszalando/ecommerceshop/order/service/OrderService.java

```diff
@@ -0,0 +1,118 @@
+package com.startstepszalando.ecommerceshop.order.service;
+
+import com.startstepszalando.ecommerceshop.cart.model.Cart;
+import com.startstepszalando.ecommerceshop.cart.model.CartItem;
+import com.startstepszalando.ecommerceshop.cart.service.CartService;
+import com.startstepszalando.ecommerceshop.exception.cart.EmptyCartException;
+import com.startstepszalando.ecommerceshop.exception.order.OrderNotFoundException;
+import com.startstepszalando.ecommerceshop.exception.product.InsufficientStockException;
+import com.startstepszalando.ecommerceshop.exception.product.ProductNotFoundException;
+import com.startstepszalando.ecommerceshop.exception.user.UserNotFoundException;
+import com.startstepszalando.ecommerceshop.order.dto.OrderProductResponse;
+import com.startstepszalando.ecommerceshop.order.dto.OrderResponse;
+import com.startstepszalando.ecommerceshop.order.model.Order;
+import com.startstepszalando.ecommerceshop.order.model.OrderProduct;
+import com.startstepszalando.ecommerceshop.order.model.OrderStatus;
+import com.startstepszalando.ecommerceshop.order.repository.OrderProductRepository;
+import com.startstepszalando.ecommerceshop.order.repository.OrderRepository;
+import com.startstepszalando.ecommerceshop.product.model.Product;
+import com.startstepszalando.ecommerceshop.product.service.ProductService;
+import com.startstepszalando.ecommerceshop.user.model.User;
+import com.startstepszalando.ecommerceshop.user.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class OrderService {
+
+    @Autowired
+    private CartService cartService;
```

Comment:
Refactor this class to use constructor injection. It's going to be a lot easier to test and is less error prone 😄